File: nvxapp.server.service/ClientServer-Service/ModelsBase/CheckObj.cs
namespace nvxapp.server.service.ClientServer_Service.ModelsBase
{
    public class CheckObjOn_Id_Text : ICheckObj<string>
    {
        public string Id { get; set; } = string.Empty;
        public bool Checked { get; set; }
    }

    public class CheckObjOn_Id_Number : ICheckObj<int>
    {
        public int Id { get; set; } = 0;
        public bool Checked { get; set; }
    }


    public interface ICheckObj<T>
    {
        T Id { get; set; }
        bool Checked { get; set; }
    }
}

File: nvxapp.client/src/nvx/ClientServer-Service/ModelsBase/check-obj.ts



export class CheckObjOn_Id_Text implements ICheckObj<string> {
  public id: string;
  public checked: boolean;
}

export class CheckObjOn_Id_Number implements ICheckObj<number> {
  public id: number;
  public checked: boolean;
}


export interface ICheckObj<T> {
  id: T;
  checked: boolean;
}
